kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -224,6 +224,10 @@ def _create_kubelet_prometheus_instance(self, instance):
         return kubelet_instance
 
     def check(self, instance):
+        # Kubelet credential defaults are determined dynamically during every
+        # check run so we must make sure that configuration is always reset
+        self._http_handlers.clear()
```

Comment:
We aren't using that private attribute anywhere else than in `OpenMetricsScraperMixin`. Should we move this to a `.clear_http_handlers()` public method there, with a docstring perhaps explaining in what situations you may want to clear those?